app/build.gradle

```diff
@@ -45,40 +54,59 @@ android {
     }
 
     buildTypes {
+        applicationVariants.all { variant ->
+            variant.buildConfigField "String", "AMAZON_API_KEY", "\"" + getApiKey() + "\""
+        }
+
         debug {
             ext.enableCrashlytics = false
         }
+
+        release {
+            minifyEnabled true
+        }
     }
 }
 
+configurations {
+    all*.exclude group: 'io.reactivex', module: 'rxjava'
+}
+
 dependencies {
     testCompile 'junit:junit:4.12'
     testCompile 'org.mockito:mockito-core:1.9.5'
 
-    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile fileTree(include: ['*.jar'], dir: 'libs')
 
-    compile 'com.android.support:support-v4:22.2.0'
-    compile 'com.android.support:appcompat-v7:22.2.0'
-    compile 'com.android.support:recyclerview-v7:22.2.0'
-    compile 'com.android.support:design:22.2.0'
+    compile 'com.android.support:support-v4:23.1.1'
+    compile 'com.android.support:appcompat-v7:23.1.1'
+    compile 'com.android.support:recyclerview-v7:23.1.1'
+    compile 'com.android.support:cardview-v7:23.1.1'
+    compile 'com.android.support:design:23.1.1'
+
+    compile 'com.google.code.gson:gson:2.4'
 
-    compile 'com.google.code.gson:gson:2.3.1'
     compile 'com.github.mazurio:glacier:0.0.3'
     compile 'net.danlew:android.joda:2.7.1'
-
     compile 'commons-io:commons-io:2.4'
     compile 'com.jakewharton:butterknife:6.1.0'
+    compile 'com.squareup.picasso:picasso:2.5.2'
 
-    compile 'com.netflix.rxjava:rxjava-core:0.20.7'
-    compile 'com.netflix.rxjava:rxjava-android:0.20.7'
+    compile 'pl.droidsonroids.gif:android-gif-drawable:1.1.1'
```

Comment:
`gif-drawable` and `picasso` should be replaced with `Glide` later on (at the moment Glide has some problems with gifs that I cannot solve).